rfd/0097-teleport-connect-usage-metrics.md

```diff
@@ -0,0 +1,221 @@
+
+#### What features are the most popular?
+
+Usage of each feature will be measured basing on events from the [events](#events) section. They will allow to generate
+various statistics, like the most common opened kinds of tabs or just show the usage of particular UI elements.
+
+#### Where users struggle the most?
+
+//TODO
+
+#### What platforms are the most popular?
+
+This will be measured in two ways.
+
+- Based on downloads count for each platform.
+- Based on a real usage. Connect will emit _start app_ event that contains OS field. These events will be then
+  aggregated by unique users.
+
+#### How usage grows or shrinks over time?
+
+PostHog allows to create [Trends](https://posthog.com/manual/trends) basing on DAU. It can be used to show how usage
+changes in a given period of time.
+
+## Events
+
+### `connect.start`
+
+Starting the app.
+
+Event properties:
+
+- `os`: one of `mac`/`windows`/`linux`
+- `connect_version`: string
+
+### `connect.cluster.add`
+
+Successful adding a cluster.
+
+Event properties:
+
+- `root_cluster_name`: string - name of the added cluster
+
+### `connect.cluster.login`
+
+Successful login to a cluster.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `connector_type`: string
+
+### `connect.workspace.change`
+
+Changing a workspace.
+
+Event properties:
+
+- `target_root_cluster_name`: string
+
+### `connect.leafClusterPicker.change`
+
+Changing a leaf cluster.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `target_leaf_cluster_name`: string
+
+### `connect.tab.open`
+
+Opening a tab.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string (optional)
+- `tab_kind`: one
+  of `resource_list`/`terminal_ssh`/`terminal_kube`/`terminal_shell`/`db_proxy`/`create_access_request`/`review_access_request`
+
+### `connect.connectionsPicker.item.select`
+
+Selecting an item from the connections picker.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string (optional)
+- `connection_kind`: one of `ssh`/`kube`/`db`
+
+### `connect.commandBar.command.run`
+
+Running a command from command bar.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string (optional)
+
+### `connect.commandBbar.autocomplete.select`
+
+Selecting autocomplete item from the command bar suggestions.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string
+- `suggestion`: one of `ssh_login`/`ssh_server`/`database`
+
+### `connect.resourceTable.search`
+
+Performing a resource search.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string
+- `resource_kind`: one of `ssh`/`kube`/`db`
+- `is_advanced`: boolean
+
+### `connect.accessRequests.create`
+
+Creating an access request.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string (optional)
+- `requested_resources_count`: number
```

Comment:
I'm not sure what are the exact plans for access requests overall but maybe we could also capture whether it's a resource vs role access request?

I suppose the `requested_resources_count` wouldn't be applicable to role access requests.

Reply:
>maybe we could also capture whether it's a resource vs role access request?

Added